mlrun/api/main.py

```diff
@@ -51,6 +55,78 @@ async def http_status_error_handler(
     )
 
 
+def get_client_addr(scope):
+    # uvicorn expects this to be a tuple while starlette test client sets it to be a list
+    if isinstance(scope.get("client"), list):
+        scope["client"] = tuple(scope.get("client"))
+    return uvicorn.protocols.utils.get_client_addr(scope)
+
+
+@app.middleware("http")
+async def log_request_response(request: fastapi.Request, call_next):
+    request_id = str(uuid.uuid4())
+    silent_logging_paths = [
+        "healthz",
+    ]
+    path_with_query_string = uvicorn.protocols.utils.get_path_with_query_string(
+        request.scope
+    )
+    if not any(
+        silent_logging_path in path_with_query_string
+        for silent_logging_path in silent_logging_paths
+    ):
+        logger.info(
+            "Received request",
+            method=request.method,
+            client_address=get_client_addr(request.scope),
+            http_version=request.scope["http_version"],
+            request_id=request_id,
+            uri=path_with_query_string,
+        )
+    try:
+        response = await call_next(request)
+
+    # Using StarletteHTTPException because
+    # https://fastapi.tiangolo.com/tutorial/handling-errors/#fastapis-httpexception-vs-starlettes-httpexception
+    except StarletteHTTPException as exc:
+        logger.warning(
+            "Request handling returned error status. Sending response",
+            status_code=exc.status_code,
+            request_id=request_id,
+            uri=path_with_query_string,
+            method=request.method,
+        )
+        raise
+    except Exception:
+        logger.warning(
+            "Request handling failed. Sending response",
+            # TODO: When finding a better solution for this turn off access logs (access_log=False in uvicorn.run call)
```

Comment:
I think you can turn off access logs and only log here. 
What info does the access-log log that you don't have here, at the point where the handler failed?
Also, what customer exception handlers is mlrun using? Can you just take care that when an exception handler is added it will properly log and then you can turn off access logs?

Reply:
Apparently I understood it the opposite way, exception handlers runs before the user supplied middleware, therefore we're good, changed stuff respectively